pkgs/applications/audio/carla/default.nix

```
@@ -44,6 +44,16 @@ stdenv.mkDerivation rec {
 
   installFlags = [ "PREFIX=$(out)" ];
 
+  patchPhase = ''
+    # --with-appname="$0" is evaluated with $0=.carla-wrapped instead of carla. Fix that.
+    for file in $(grep -rl -- '--with-appname="$0"' | grep -v "test.sh")
+    do
+        echo "Patching $file"
+        filename="$(basename -- "$file")"
+        sed -i "s/--with-appname=\"\$0\"/--with-appname=\"$filename\"/g" "$file"
```

use substitute in place so something like this

```suggestion
    substituteInPlace $file \
      --replace '--with-appname="$0"' '--with-appname="$filename"'
```

---

Thanks, modified! (there were a small issue with escaping)